client/post-editor/editor-revisions/dialog.jsx

```diff
@@ -0,0 +1,118 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import { get, flow, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getPostRevisionsSelectedRevision, isPostRevisionsDialogVisible } from 'state/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { togglePostRevisionsDialog } from 'state/posts/revisions/actions';
+import EditorRevisions from 'post-editor/editor-revisions';
+import Dialog from 'components/dialog';
+
+class PostRevisionsDialog extends PureComponent {
+	static propTypes = {
+		/**
+		 * loadRevision is passed through from `post-editor/post-editor.jsx`
+		 * @TODO untangle & reduxify
+		 */
+		loadRevision: PropTypes.func.isRequired,
+		onClose: PropTypes.func,
+
+		// connected to state
+		isVisible: PropTypes.bool.isRequired,
+
+		// connected to dispatch
+		recordTracksEvent: PropTypes.func.isRequired,
+		toggleDialog: PropTypes.func.isRequired,
+
+		// localize
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		onClose: noop,
+	};
+
+	componentWillMount() {
+		this.toggleBodyClass( { isVisible: this.props.isVisible } );
+	}
+
+	componentWillUpdate( { isVisible } ) {
+		this.toggleBodyClass( { isVisible } );
+	}
+
+	toggleBodyClass( { isVisible } ) {
+		if ( ! ( typeof document === 'object' && get( document, 'body.classList' ) ) ) {
+			return;
+		}
+
+		const bodyClassName = 'showing-post-revisions-dialog';
+		isVisible
```

Comment:
This seems like a strange usage of the ternary operator. It's primarily designed for getting a value back, not for conditional fn calls.

Reply:
I would turn this into an `if` statement. We cannot use `classList.toggle` as we need a support for IE 11 and they never implemented it there, only in Edge :/

Reply:
An `if` could be a little more clear. I'm a proponent for this use of ternaries though - I feel like people are becoming much more used to ternaries than they were a year or two ago.
In terms of `classList.toggle` - I'm not a fan of that anyway. In fact I'm not a fan of `toggle` in general as I'd always prefer to explicitly add or remove. Particularly when toggle takes an argument... a true toggle should flip in every case without influence IMO.